src/io/mod.rs

```diff
@@ -101,20 +116,23 @@ impl Io {
 
         if Self::is_gzip_path(p) {
             Ok(Box::new(BufReader::with_capacity(self.buffer_size, MultiGzDecoder::new(buf))))
+        } else if Self::is_zstd_path(p) {
+            Ok(Box::new(BufReader::with_capacity(self.buffer_size, ZstdDecoder::new(buf).unwrap())))
         } else {
             Ok(Box::new(buf))
         }
     }
 
-    /// Opens a file for writing. Transparently handles writing GZIP'd data if the file
-    /// ends with a recognized GZIP extension.
+    /// Opens a file for writing. Transparently handles encoding data in gzip and zstd formats.
     pub fn new_writer<P>(&self, p: &P) -> Result<BufWriter<Box<dyn Write + Send>>>
     where
         P: AsRef<Path>,
     {
         let file = File::create(p).map_err(FgError::IoError)?;
         let write: Box<dyn Write + Send> = if Io::is_gzip_path(p) {
             Box::new(GzEncoder::new(file, self.compression))
+        } else if Io::is_zstd_path(p) {
+            Box::new(ZstdEncoder::new(file, 0).unwrap().auto_finish())
```

Comment:
Same comment here about the `unwrap()`.

Reply:
How about something like:

![Screenshot 2023-08-18 at 12 26 25 PM](https://github.com/fulcrumgenomics/fgoxide/assets/122313882/3cc1acc4-ee67-493d-9045-0cba5eef3c05)

Not sure if it's the proper way to go about it but it's one possibility if it's preferred that the Io struct absolutely doesn't change.

Reply:
I could see having a member per level, but then what about bzip2 and future compression levels?  And perhaps we have compression that requires not just a level?  And certainly we don't want to add translation layer that goes from 0-10 levels for each compression type.

What about instead of storing `compression` as a member of `Io`, we have a Map/Vec that stores a compression-specific struct (can impl a trait).  You could even make `is_gzip_path` specific to the compression as a trait-level method (`is_path_for`), so a new compression need only impl the base trait to be added.  Then we avoid the if/else if/else chain there and it can just be a loop (with a default if it exits the loop)?  Does that make sense?

Reply:
Makes sense. The `Io` struct probably will have to change at some point if further compression schemes are to be supported as far as I can see. #10 might also be a good solution to this. Haven't checked it out thoroughly yet, but it does feel like good idea to actually check the file format instead of just the extension, although might still want to have a way to compare the found format vs. extension to detect any inconsistencies.